packages/html/ds/src/tabs/tabs.ts

```diff
@@ -0,0 +1,189 @@
+import {
+  BaseComponent,
+  BaseComponentOptions,
+  initialiseModule,
+} from '../common/component';
+
+export type TabsOptions = BaseComponentOptions;
+
+class TabsAutomatic {
+  tabs: HTMLInputElement[];
+  firstTab!: HTMLInputElement;
+  lastTab!: HTMLInputElement;
+  tabElement: Element;
+  tabPanels: HTMLElement[];
+
+  constructor(tabElement: Element) {
+    this.tabElement = tabElement;
+
+    this.tabs = [];
+
+    this.tabs = [
+      ...this.tabElement.querySelectorAll('[role=tab]'),
+    ] as HTMLInputElement[];
+    this.tabPanels = [];
+
+    for (const tab of this.tabs) {
+      if (!this.firstTab) {
+        this.firstTab = tab;
+      }
+      this.lastTab = tab;
+
+      const ariaControlAttribute = tab.getAttribute('aria-controls');
+      if (!ariaControlAttribute) {
+        continue;
+      }
+
+      const tabpanel = document.querySelector(
+        `#${ariaControlAttribute}`,
+      ) as HTMLElement;
+
+      if (tabpanel) {
+        tab.tabIndex = -1;
+        tab.setAttribute('aria-selected', 'false');
+        this.tabPanels.push(tabpanel);
+
+        tab.addEventListener('keydown', this.onKeydown.bind(this));
+        tab.addEventListener('click', this.onClick.bind(this));
+      }
+    }
+
+    this.setSelectedTab(this.firstTab, false);
+  }
+
+  attachListeners() {
```

Comment:
Can we keep the same naming conventions across the components ? So far we've been using `initComponent` and `destroyComponent`